src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -162,103 +167,131 @@ public void validateIncompatibleJavadocCheckNames() throws IOException {
 
     @Test
     public void validateIntegrationTestClassNames() throws Exception {
-        final Set<String> compatibleChecks = new HashSet<>();
-        final Pattern pattern = Pattern.compile("^XpathRegression(.+)Test\\.java$");
-        try (DirectoryStream<Path> javaPaths = Files.newDirectoryStream(javaDir)) {
-            for (Path path : javaPaths) {
-                assertWithMessage(path + " is not a regular file")
-                        .that(Files.isRegularFile(path))
-                        .isTrue();
-                final String filename = path.toFile().getName();
-                if (filename.startsWith("Abstract")) {
-                    continue;
-                }
-
-                final Matcher matcher = pattern.matcher(filename);
-                assertWithMessage(
-                            "Invalid test file: " + filename + ", expected pattern: " + pattern)
-                        .that(matcher.matches())
-                        .isTrue();
-
-                final String check = matcher.group(1);
-                assertWithMessage("Unknown check '" + check + "' in test file: " + filename)
-                        .that(SIMPLE_CHECK_NAMES)
-                        .contains(check);
-
-                assertWithMessage(
-                            "Check '" + check + "' is now compatible with SuppressionXpathFilter."
-                                + " Please update the todo list in"
-                                + " XpathRegressionTest.INCOMPATIBLE_CHECK_NAMES")
-                        .that(INCOMPATIBLE_CHECK_NAMES.contains(check))
-                        .isFalse();
-                compatibleChecks.add(check);
-            }
-        }
-
         // Ensure that all lists are up-to-date
         final Set<String> allChecks = new HashSet<>(SIMPLE_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_JAVADOC_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_CHECK_NAMES);
         allChecks.removeAll(Set.of("Regexp", "RegexpSinglelineJava", "NoCodeInFile"));
         allChecks.removeAll(NO_VIOLATION_MODULES);
-        allChecks.removeAll(compatibleChecks);
+        allChecks.removeAll(compatibleChecks());
         allChecks.removeAll(INTERNAL_MODULES);
 
         assertWithMessage("XpathRegressionTest is missing for [" + String.join(", ", allChecks)
                 + "]. Please add them to src/it/java/org/checkstyle/suppressionxpathfilter")
-                        .that(allChecks)
-                        .isEmpty();
+                .that(allChecks)
+                .isEmpty();
     }
 
+    private Set<String> compatibleChecks() throws IOException {
+        try (Stream<Path> javaPaths = Files.list(javaDir)) {
+            return javaPaths
+                    .filter(Files::isRegularFile)
```

Comment:
This reads like a story. `final String filename =` is not  a real english wording as its only for computer communication syntax. Using the functional clean code approach removing all overhead leaving only the pure commands and code we need to declare.